src/storage/txn/scheduler.rs

```diff
@@ -471,33 +483,30 @@ fn process_read<E: Engine>(
                 None,
                 ctx.get_isolation_level(),
             );
-            let res = match reader.seek_ts(start_ts).map_err(StorageError::from) {
-                Err(e) => ProcessResult::Failed { err: e },
-                Ok(opt) => match opt {
-                    Some(key) => match find_mvcc_infos_by_key(&mut reader, &key, u64::MAX) {
-                        Ok((lock, writes, values)) => ProcessResult::MvccStartTs {
-                            mvcc: Some((
-                                key,
-                                MvccInfo {
-                                    lock,
-                                    writes,
-                                    values,
-                                },
-                            )),
-                        },
-                        Err(e) => ProcessResult::Failed { err: e.into() },
-                    },
-                    None => ProcessResult::MvccStartTs { mvcc: None },
-                },
-            };
-            statistics.add(reader.get_statistics());
-            res
+            match reader.seek_ts(start_ts)? {
+                Some(key) => {
+                    let (lock, writes, values) =
+                        find_mvcc_infos_by_key(&mut reader, &key, u64::MAX)?;
+                    statistics.add(reader.get_statistics());
```

Comment:
Is it OK to skip statistics when error happens?

Reply:
Thanks! Updated. Now statistics is collected even if there are errors.